cmd/erasure-metadata.go

```diff
@@ -365,57 +312,80 @@ func getXLMetaInQuorum(ctx context.Context, metaArr []xlMetaV1, modTime time.Tim
 		}
 	}
 
-	return xlMetaV1{}, errXLReadQuorum
+	return FileInfo{}, errErasureReadQuorum
 }
 
-// pickValidXLMeta - picks one valid xlMeta content and returns from a
-// slice of xlmeta content.
-func pickValidXLMeta(ctx context.Context, metaArr []xlMetaV1, modTime time.Time, quorum int) (xmv xlMetaV1, e error) {
-	return getXLMetaInQuorum(ctx, metaArr, modTime, quorum)
+// pickValidFileInfo - picks one valid FileInfo content and returns from a
+// slice of ermeta content.
+func pickValidFileInfo(ctx context.Context, metaArr []FileInfo, modTime time.Time, quorum int) (xmv FileInfo, e error) {
+	return getFileInfoInQuorum(ctx, metaArr, modTime, quorum)
 }
 
 // list of all errors that can be ignored in a metadata operation.
 var objMetadataOpIgnoredErrs = append(baseIgnoredErrs, errDiskAccessDenied, errVolumeNotFound, errFileNotFound, errFileAccessDenied, errCorruptedFormat)
 
-// writeXLMetadata - writes `xl.json` to a single disk.
-func writeXLMetadata(ctx context.Context, disk StorageAPI, bucket, prefix string, xlMeta xlMetaV1) error {
-	jsonFile := path.Join(prefix, xlMetaJSONFile)
+// Rename metadata content to destination location for each disk in order.
```

Comment:
```suggestion
// Rename metadata content to destination location for each disk.
// If quorum cannot be reached the function will automatically attempt 
// to undo the rename on successful disks.
```

It is done concurrently, so no order AFAICT.